src/execution/operator/aggregate/physical_window.cpp

```diff
@@ -76,17 +76,17 @@ class WindowHashGroup {
 				stage = WindowGroupStage::FINALIZE;
 				return true;
 			}
-			return false;
+			break;
 		case WindowGroupStage::FINALIZE:
 			if (finalized == blocks) {
 				stage = WindowGroupStage::GETDATA;
 				return true;
 			}
-			return false;
+			break;
 		default:
-			// never block in GETDATA
```

Comment:
We're still never blocking here - and if I modify this to block then I get deadlocks again. Perhaps we should leave this as-is and leave the comment as well?

Reply:
For future reference - the way I tested the deadlocks is with this diff:

```diff
diff --git a/benchmark/include/benchmark.hpp b/benchmark/include/benchmark.hpp
index ece2dda3d3..acd2f3d887 100644
--- a/benchmark/include/benchmark.hpp
+++ b/benchmark/include/benchmark.hpp
@@ -28,7 +28,7 @@ struct BenchmarkState {
 //! The base Benchmark class is a base class that is used to create and register
 //! new benchmarks
 class Benchmark {
-       constexpr static size_t DEFAULT_NRUNS = 5;
+       constexpr static size_t DEFAULT_NRUNS = 500;
        Benchmark(Benchmark &) = delete;
 
 public:
diff --git a/src/execution/operator/aggregate/physical_window.cpp b/src/execution/operator/aggregate/physical_window.cpp
index 9435c24de7..9f9341fcdb 100644
--- a/src/execution/operator/aggregate/physical_window.cpp
+++ b/src/execution/operator/aggregate/physical_window.cpp
@@ -84,7 +84,7 @@ public:
                        }
                        break;
                default:
-                       return true;
+                       break;
                }
                return false;
        }

```

Then running:

```
build/reldebug/benchmark/benchmark_runner benchmark/tpcds/sf1/q49.benchmark
```

Which gets stuck for me spuriously in one of the hundreds of runs.